std/encoding.d

```diff
@@ -2250,24 +2138,27 @@ body
             immutable minReservePlace = 6;
         }
 
-        Dst[] buffer = new Dst[s.length];
```

Comment:
The performance difference could be negligible for all I know, I haven't measured it. I would have thought most of std.encoding was pure already, but if it's extra work then I think it's better to leave it to the future.